flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/DataStatisticsCoordinator.java

```diff
@@ -91,7 +91,7 @@ class DataStatisticsCoordinator implements OperatorCoordinator {
     this.context = context;
     this.schema = schema;
     this.sortOrder = sortOrder;
-    this.comparator = SortOrderComparators.forSchema(schema, sortOrder);
+    this.comparator = Comparators.forType(SortKeyUtil.sortKeySchema(schema, sortOrder).asStruct());
```

Comment:
this is an important fix. SortKey may have been serialized by subtask and deserialized by the coordinator. we need to use the schema of the transformed sort fields for struct comparator.